pkg/controllers/management/feature/feature_handler.go

```diff
@@ -46,9 +48,51 @@ func (h *handler) sync(_ string, obj *v3.Feature) (*v3.Feature, error) {
 	if obj.Name == features.Harvester.Name() {
 		return obj, h.toggleHarvesterNodeDriver(obj.Name)
 	}
+
+	if obj.Name == features.HarvesterBaremetalContainerWorkload.Name() {
+		return obj, h.syncHarvesterFeature(obj)
+	}
 	return obj, nil
 }
 
+// syncHarvesterFeature ensures that Harvester feature is enabled
+// if baremetal management feature is enabled and annotates feature with experimental annotation
+func (h *handler) syncHarvesterFeature(obj *v3.Feature) error {
+
+	objCopy := obj.DeepCopy()
+
+	if objCopy.Annotations == nil {
+		objCopy.Annotations = make(map[string]string)
+	}
+
+	if val, ok := objCopy.Annotations[v3.ExperimentalFeatureKey]; !ok || val != v3.ExperimentalFeatureValue {
+		objCopy.Annotations[v3.ExperimentalFeatureKey] = v3.ExperimentalFeatureValue
+	}
+
+	if !reflect.DeepEqual(obj, objCopy) {
+		_, err := h.featuresClient.Update(objCopy)
+		return err
```

Comment:
This is just a minor thing, but should we wrap this error? The reason I ask is because we're making calls from `h.featuresClient` in multiple places within this function, and if any of those calls returns an error, we'll return that error without wrapping. This makes it difficult to figure out which of the calls from `h.featuresClient` returned the error. If you wrap each error with a different message, then it's immediately clear which call produced the error.

For example, you could wrap this error as follows.
```suggestion
		return fmt.Errorf("error updating Feature object: %w", err)
```

Then below you can do
```go
		if !reflect.DeepEqual(harvesterFeature, harvesterFeatureCopy) {
			_, err := h.featuresClient.Update(harvesterFeatureCopy)
			return fmt.Errorf("error updating Harvester feature object: %w", err)
		}
```